src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -191,7 +194,39 @@ public CallOutput EstimateGas(BlockHeader header, Transaction tx, CancellationTo
                 estimateGasTracer.WithCancellation(cancellationToken));
             
             long estimate = estimateGasTracer.CalculateEstimate(tx);
-            return new CallOutput {Error = tryCallResult.Success ? estimateGasTracer.Error : tryCallResult.Error, GasSpent = estimate, InputError = !tryCallResult.Success};
+            
+            return new CallOutput 
+            {
+                Error = tryCallResult.Success ? estimateGasTracer.Error : tryCallResult.Error, 
+                GasSpent = estimate, 
+                InputError = !tryCallResult.Success
+            };
+        }
+
+        public CallOutput CreateAccessList(BlockHeader header, Transaction tx, CancellationToken cancellationToken, bool optimize)
+        {
+            CallOutputTracer callOutputTracer = new();
+            AccessTxTracer accessTxTracer = optimize 
+                ? new(tx.SenderAddress, 
+                    tx.GetRecipient(tx.IsContractCreation ? _stateReader.GetNonce(header.StateRoot, tx.SenderAddress) : 0)) 
+                : new();
+
+            (bool Success, string Error) tryCallResult = TryCallAndRestore(header, header.Number, header.Timestamp, tx,
+                new CancellationTxTracer(new CompositeTxTracer(callOutputTracer, accessTxTracer), cancellationToken)
```

Comment:
.WithCancellation

Reply:
Yes but we want to specify on what we are cancelling, I am letting it slide for now. Maybe if we have more of them.